notifications/source/client.go

```diff
@@ -0,0 +1,196 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create HTTP request")
+	}
+
+	ruleIdsStrArr := make([]string, 0, len(ruleIDs))
+	for _, ruleId := range ruleIDs {
+		ruleIdsStrArr = append(ruleIdsStrArr, strconv.FormatInt(ruleId, 10))
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add(XIcingaRulesVersion, ruleVersion)
+	req.Header.Add(XIcingaRulesId, strings.Join(ruleIdsStrArr, ","))
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot POST HTTP request to process event")
+	}
+
+	defer func() {
+		_, _ = io.Copy(io.Discard, resp.Body)
+		_ = resp.Body.Close()
+	}()
+
+	if resp.StatusCode == http.StatusPreconditionFailed {
+		// Indicates that the rules version is outdated and the body should contain the current rules and their version.
+		// So, we read the body to extract the rules and return an ErrRulesOutdated error, so the caller can retry
+		// the event submission after it has reevaluated them.
+		var rulesInfo RulesInfo
+		if err := json.NewDecoder(resp.Body).Decode(&rulesInfo); err != nil {
+			return nil, errors.Wrap(err, "cannot decode new rules from process event response")
+		}
+
+		return &rulesInfo, ErrRulesOutdated
+	}
+
+	if resp.StatusCode >= http.StatusOK && resp.StatusCode <= 299 {
+		return nil, nil // Successfully processed the event.
+	}
+
+	if resp.StatusCode == http.StatusNotAcceptable {
+		return nil, nil // Superfluous state change event.
+	}
+
+	var buf bytes.Buffer
+	_, _ = io.Copy(&buf, &io.LimitedReader{R: resp.Body, N: 1 << 20}) // Limit the error message length to avoid memory exhaustion.
+
+	return nil, errors.Errorf("unexpected response from process event API, status %q (%d): %q",
+		resp.Status, resp.StatusCode, strings.TrimSpace(buf.String()))
+}
+
+// JoinIcingaWeb2Path constructs a URL by joining the Icinga Web 2 base URL with the provided relative URL.
+//
+// It is used to convert any relative URL into an absolute URL that points to the Icinga Web 2 instance.
+// A relative URL like "/icingadb/host" is transformed to e.g. "https://icinga.example.com/icingaweb2/icingadb/host"
+// after passing through this method, assuming the Icinga Web 2 base URL is "https://icinga.example.com/icingaweb2".
+func (c *Client) JoinIcingaWeb2Path(relativePath string) *url.URL {
+	return c.IcingaWebBaseUrl.JoinPath(relativePath)
+}
+
+// EmptyRulesVersion is a constant representing the version of the rules when no rules are present.
+// It is used to indicate that there are no rules available for a given source.
+const EmptyRulesVersion = "0x0"
+
+// RulesInfo holds information about the event rules for a specific source.
+type RulesInfo struct {
+	Version string             // Version of the event rules fetched from the API.
+	Rules   map[int64]RuleResp // Rules is a map of rule IDs to their corresponding RuleResp objects.
+}
+
+// Iter returns an iterator over the rules in the SourceRulesInfo.
```

Comment:
```suggestion
// Iter returns an iterator over the rules in the RulesInfo.
```

Reply:
This code won't last long though ;)! When sources start to manage their own event rules config by their own, this will become obsolete.